src/fontinfo.rs

```diff
@@ -0,0 +1,940 @@
+use serde::de::Deserializer;
+use serde::ser::{SerializeSeq, Serializer};
+use serde::{Deserialize, Serialize};
+
+use crate::shared_types::Guideline;
+use crate::Error;
+
+// The specification is vague about data type limits, usually implicitly meaning
+// Python types. Since Python is dynamic, the spec does not nail down the exact type
+// in several locations and we have to assume the biggest type that can hold it.
+type Integer = i32;
+type NonNegativeInteger = u32;
+type IntegerOrFloat = f64;
```

Comment:
again the 'strictly most correct' way to deal with this would be to make it an enum with `Float` and `Integer` variants; but I suspect in practice the integers we're dealing with are in the range that can be fully represented by an `f64`. 🤷‍♀

Reply:
I think so. There aren't even any 64 bit values in the spec that I know of.